src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,119 @@
+        String findByIdSqlQuery = "SELECT * FROM book WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(findByIdSqlQuery)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(getBook(resultSet));
+            }
+            return Optional.empty();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find book by id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        String findAllSqlQuery = "SELECT * FROM book";
+        List<Book> books = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(findAllSqlQuery)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                books.add(getBook(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find all books", e);
+        }
+        return books;
+    }
+
+    @Override
+    public Book update(Book book) {
+        String updateSqlQuery = "UPDATE book SET title = ?, price = ? WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement =
+                        connection.prepareStatement(updateSqlQuery)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.setLong(3, book.getId());
+            checkAffectedRows(statement.executeUpdate());
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update book" + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String deleteSqlQuery = "DELETE FROM book WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement =
+                        connection.prepareStatement(deleteSqlQuery,
+                                Statement.RETURN_GENERATED_KEYS)) {
+            statement.setLong(1, id);
+            int affectedRows = statement.executeUpdate();
+            return affectedRows > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete book by id " + id, e);
+        }
+    }
+
+    private static void checkAffectedRows(int affectedRows) {
+        if (affectedRows < 1) {
+            throw new RuntimeException("Expected to change at least 1 row, but was "
+                    + affectedRows);
+        }
+    }
+
+    private Book getBook(ResultSet resultSet) throws SQLException {
+        Long id = resultSet.getLong(ID);
```

Comment:
use `getObject(ID, Long.class)` instead, just to be sure and in accordance to common mistakes